### ISSUE TITLE
NS107S support

### ISSUE THREAD

**Issue by laforestlabs** [FastLED/FastLED#843]
It look like new star is making a [new LED they are calling NS107S](https://cdn.hackaday.io/files/21293910999712/NS107S%20RGB%20LED%20Chip.pdf). This is an interesting option because the PWM rate is very high at 26 khz, making it useful for POV applications. Also it look like it comes in small size packages like 3535 and 2020. 

Is there any chance FastLED will support this LED?

**Comment by focalintent** [comment]
The protocol looks identical to the APA102 - so nothing will probably need to be done to support it.

(Given the failure of the APA102 and SK clones to knock the clockless WS281x style chipsets off the throne, I suspect this will stay a niche led player, though)

**Comment by focalintent** [comment]
Ah - it's just the APA107 which is the successor to the APA102. I wonder if they've fixed the signal degradation issue with the APA102 that requires lowering the clock speed the longer your strip of leds is...

**Comment by dimitre** [comment]
Yes you can use the same protocol and I think they fixed the problem described here by Paul
https://www.pjrc.com/why-apa102-leds-have-trouble-at-24-mhz/

**Comment by zackees** [comment]
Closing — **already supported**. @focalintent identified it correctly within minutes, and @dimitre confirmed it in practice:

> it's just the APA107 which is the successor to the APA102

> Yes you can use the same protocol

The NS107S speaks the APA102 protocol, so no new chipset type is needed. Use `APA102`:

```cpp
FastLED.addLeds<APA102, DATA_PIN, CLOCK_PIN, BGR>(leds, NUM_LEDS);
```

The whole APA102 family is available — `APA102`, `APA102HD`, `SK9822`, `SK9822HD`, `DOTSTAR`, `DOTSTARHD`, `HD107HD`. The `HD` variants add 5-bit gamma correction, which is worth trying here: the NS107S's high PWM rate is most useful at low brightness, and that's exactly where the extra gamma resolution shows.

**On the clock-speed question** @focalintent and @dimitre raised — [Paul Stoffregen's write-up](https://www.pjrc.com/why-apa102-leds-have-trouble-at-24-mhz/) on APA102 signal degradation over long strips — FastLED lets you set the SPI data rate explicitly:

```cpp
FastLED.addLeds<APA102, DATA_PIN, CLOCK_PIN, BGR, DATA_RATE_MHZ(12)>(leds, NUM_LEDS);
```

So if your NS107S strip does turn out to tolerate a higher clock than an APA102 (dimitre thought they'd fixed it), you can just raise it, and drop it if a long run misbehaves. There's also `HD107HD`, which exists specifically as a 40 MHz turbo variant of the same protocol — a useful reference point for how far the family can be pushed.

**For POV specifically** — which is why you were interested — the data rate is usually the binding constraint rather than the LED's PWM rate, so being able to tune `DATA_RATE_MHZ` matters more than the chipset name.

@focalintent's aside has aged well, incidentally: the clocked APA102-family parts did stay niche next to the WS281x clockless chipsets. But they remain the right choice for POV, and they're fully supported.

Nothing to implement here, so closing. If an NS107S turns out to need timing that genuinely differs from APA102, please open a fresh issue with the specifics — the datasheet link in your original post still resolves, which is more than most chipset issues from this era can say.
